### PR TITLE
[eda][build] dependencies cleanup, CI stability fixes, removed unused code

### PR DESCRIPTION
**Pull request by gradientsky**
*Issue:* #2846 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eda/setup.py [hunk 1]
```diff
@@ -20,16 +20,15 @@
...
-    'matplotlib>3.3',
+    'numpy',  # version range defined in `core/_setup_utils.py`
+    'scipy',  # version range defined in `core/_setup_utils.py`
+    'scikit-learn',  # version range defined in `core/_setup_utils.py`
+    'pandas',  # version range defined in `core/_setup_utils.py`
+    'matplotlib>3.3,<3.7',
     'missingno>=0.5.1,<0.6',
     'phik>=0.12.2,<0.13',
     'seaborn>=0.12.0,<0.13',
-    'ipython>=7.34,<9.0',
```
Comment (Innixma) [inline eda/setup.py:31]: can you explain why you removed ipython?

**Review by Innixma — APPROVED** [review]
LGTM! Had 1 comment

File: eda/setup.py (hunk 1 above)
Comment (gradientsky) [reply eda/setup.py:31]: `ipywidget` dependency already have a transitive dependency to `ipython`

Comment (Innixma) [reply eda/setup.py:31]: Gotcha, thanks!

**Pull request merged**